aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -0,0 +1,237 @@
+module AptosFramework::Stake {
+    use Std::Vector;
+    use Std::Signer;
+    use Std::Option::{Self, Option};
+    use CoreFramework::SystemAddresses;
+    use CoreFramework::Timestamp;
+//    use CoreFramework::Reconfiguration;
+    use AptosFramework::TestCoin::{Self, Coin};
+
+    const MINIMUM_LOCK_PERIOD: u64 = 86400;
+    const MINIMUM_RECONFIG_PERIOD: u64 = 3600;
+
+    /// Basic unit of stake delegation, it's stored in StakeInfo.
+    struct Delegation has store {
+        coin: Coin,
+        from: address,
+        locked_until_secs: u64,
+    }
+
+    /// Aggregation of delegation and represent a validator's voting power, stored in ValidatorInfo.
+    struct StakeInfo has store {
+        // sum of active and cooldown stakes, updated on epoch boundary.
+        current_stake: u64,
+        // active stake
+        active: vector<Delegation>,
+        // pending activation for next epoch
+        warmup: vector<Delegation>,
+        // pending inactivation for next epoch
+        cooldown: vector<Delegation>,
+        // inactive stake, can be withdrawn
+        inactive: vector<Delegation>,
+    }
+
+    /// Consensus information per validator, stored in validator address.
+    struct ValidatorInfo has key {
+        stake_info: StakeInfo,
+        consensus_pubkey: vector<u8>,
+        network_address: vector<u8>,
+    }
+
+    /// Full ValidatorSet, stored in @CoreResource.
+    struct ValidatorSet has key {
+        consensus_scheme: u8,
+        minimum_stake: u64,
+        maximum_stake: u64,
+        // should we store it in place?
+        validators: vector<address>,
+        last_update_time_secs: u64,
+    }
+
+    /// Any user can delegate a stake.
+    fun delegate_stake(account: &signer, to: address, coin: Coin, locked_until_secs: u64) acquires ValidatorInfo {
+        let current_time = Timestamp::now_seconds();
+        assert!(current_time + MINIMUM_LOCK_PERIOD < locked_until_secs, 0);
+        let stake_info = &mut borrow_global_mut<ValidatorInfo>(to).stake_info;
```

Comment:
What is the behavior if the `to` address is not a validator address? Should we check for that and send an appropriate error message?

Reply:
it doesn't need to be as long as it has the ValidatorInfo, because people may need to aggregate enough stakes before join the validator set